dataland-email-service/src/test/kotlin/org/dataland/datalandemailservice/services/EmailSubscriptionTrackerTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.dataland.datalandemailservice.services
+
+import jakarta.transaction.Transactional
+import org.dataland.datalandemailservice.DatalandEmailService
+import org.dataland.datalandemailservice.entities.EmailSubscriptionEntity
+import org.dataland.datalandemailservice.repositories.EmailSubscriptionRepository
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.context.SpringBootTest
+import java.util.UUID
+
+@SpringBootTest(classes = [DatalandEmailService::class], properties = ["spring.profiles.active=nodb"])
+@AutoConfigureTestDatabase(connection = EmbeddedDatabaseConnection.H2)
+@Transactional
+class EmailSubscriptionTrackerTest(
+    @Autowired private val emailSubscriptionRepository: EmailSubscriptionRepository,
+) {
+    private lateinit var emailSubscriptionTracker: EmailSubscriptionTracker
+
+    final val subscribedUuid = UUID.randomUUID()
+    final val unsubscribedUuid = UUID.randomUUID()
+
+    final val unknownEmail = "unknown@mail.com"
+    final val subscribedEmail = "subscriber@mail.com"
+    final val unsubscribedEmail = "unsubscriber@mail.com"
+
+    val subscribedEntity =
+        EmailSubscriptionEntity(
+            uuid = subscribedUuid,
+            emailAddress = subscribedEmail,
+            isSubscribed = true,
+        )
+
+    val unsubscribedEntity =
+        EmailSubscriptionEntity(
+            uuid = unsubscribedUuid,
+            emailAddress = unsubscribedEmail,
+            isSubscribed = true,
+        )
+
+    @BeforeEach
+    fun setup() {
+        emailSubscriptionTracker = EmailSubscriptionTracker(emailSubscriptionRepository)
+        emailSubscriptionRepository.deleteAll()
+        emailSubscriptionRepository.save(subscribedEntity)
+        emailSubscriptionRepository.save(unsubscribedEntity)
+    }
+
+    @Test
+    fun `validate if a new subscription entity is created for a unknown email`() {
+        val uuid = emailSubscriptionTracker.insertSubscriptionEntityIfNeededAndReturnUuid(unknownEmail)
+        val newEntity = emailSubscriptionRepository.findByEmailAddress(unknownEmail)
+        if (newEntity != null) {
```

Comment:
don't understand the if case. Shouldn't it be clear whether the newEntity is null or not? I mean, if `null` is returned here, the test run will be green although it didn't test what it was supposed to test, right? So if it's null the test should fail